python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -493,6 +507,9 @@ def get_required_multi_asset_keys(self, asset_key: AssetKey) -> AbstractSet[Asse
             return self._required_multi_asset_sets_by_key[asset_key]
         return set()
 
+    def get_required_asset_and_check_keys(self, key: SpecKey) -> AbstractSet[SpecKey]:
```

Comment:
could be a more generic name `get_required_specs_by_key`. But I think this is clearer for now, and we can change it if we add a third spec type